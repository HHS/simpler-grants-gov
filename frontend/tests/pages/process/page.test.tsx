import { render, screen, waitFor } from "@testing-library/react";
import { axe } from "jest-axe";
import { identity } from "lodash";
import Process from "src/app/[locale]/process/page";
import {
  localeParams,
  mockMessages,
  useTranslationsMock,
} from "src/utils/testing/intlMocks";

jest.mock("react", () => ({
  ...jest.requireActual<typeof import("react")>("react"),
  use: jest.fn(() => ({
    locale: "en",
  })),
}));

jest.mock("next-intl/server", () => ({
  getTranslations: () => identity,
  setRequestLocale: identity,
}));

jest.mock("next-intl", () => ({
  useTranslations: () => useTranslationsMock(),
  useMessages: () => mockMessages,
}));

describe("Process", () => {
  it("renders intro text", () => {
    render(<Process params={localeParams} />);
    const content = screen.getByText("intro.content");
    expect(content).toBeInTheDocument();
  });

  it("passes accessibility scan", async () => {
    const { container } = render(<Process params={localeParams} />);
    const results = await waitFor(() => axe(container));
    expect(results).toHaveNoViolations();
  });
});
