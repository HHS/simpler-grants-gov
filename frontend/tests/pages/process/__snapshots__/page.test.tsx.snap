// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process renders homepage unchanged 1`] = `
<div>
  <div
    class="margin-top-5"
    data-testid="beta-alert"
  >
    <div
      class="grid-container"
      data-testid="gridContainer"
    >
      <div
        class="usa-alert usa-alert--warning usa-alert--no-icon"
        data-testid="alert"
      >
        <div
          class="usa-alert__body"
        >
          <h2
            class="usa-alert__heading"
          >
            alert_title
          </h2>
          <p
            class="usa-alert__text"
          >
            alert
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="grid-container padding-top-1 tablet:padding-top-3 desktop-lg:padding-top-4"
    data-testid="breadcrumb"
  >
    <nav
      aria-label="Breadcrumbs"
      class="usa-breadcrumb"
    >
      <ol
        class="usa-breadcrumb__list"
        itemscope=""
        itemtype="https://schema.org/BreadcrumbList"
      >
        <li
          class="usa-breadcrumb__list-item"
          itemprop="itemListElement"
          itemscope=""
          itemtype="https://schema.org/ListItem"
        >
          <a
            class="usa-breadcrumb__link"
            href="/"
            itemid="/"
            itemprop="item"
            itemscope=""
            itemtype="https://schema.org/WebPage"
          >
            <span
              itemprop="name"
            >
              Home
            </span>
          </a>
          <meta
            content="1"
            itemprop="position"
          />
        </li>
        <li
          aria-current="page"
          class="usa-breadcrumb__list-item usa-current"
          itemprop="itemListElement"
          itemscope=""
          itemtype="https://schema.org/ListItem"
        >
          <span
            itemprop="name"
          >
            Process
          </span>
          <meta
            content="2"
            itemprop="position"
          />
        </li>
      </ol>
    </nav>
  </div>
  <div
    class="grid-container padding-y-1 tablet:padding-y-3 desktop-lg:padding-y-6 padding-top-0 tablet:padding-top-0 desktop-lg:padding-top-0"
    data-testid="gridContainer"
  >
    <h1
      class="margin-bottom-0 tablet-lg:font-sans-xl desktop-lg:font-sans-2xl margin-top-0"
    >
      intro.title
    </h1>
    <div
      class="grid-row grid-gap"
      data-testid="grid"
    >
      <div
        class=""
        data-testid="grid"
      >
        <p
          class="usa-intro margin-top-2"
        >
          intro.content
        </p>
      </div>
    </div>
    <div
      class="grid-row grid-gap-lg margin-top-2"
      data-testid="grid"
    >
      <div
        class="tablet-lg:grid-col-4 margin-bottom-6"
        data-testid="grid"
      >
        <div
          class="border radius-md border-base-lighter padding-x-205"
        >
          <h2
            class="tablet-lg:font-sans-lg"
          >
            intro.boxes.0.title
          </h2>
          <p
            class="margin-top-0 font-sans-md line-height-sans-4 desktop-lg:line-height-sans-6"
          >
            intro.boxes.0.content
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="padding-top-4 bg-base-lightest"
  >
    <div
      class="grid-container grid-container padding-y-4 tablet:padding-y-3 desktop-lg:padding-y-6 padding-top-0 tablet:padding-top-0 desktop-lg:padding-top-0 border-bottom-2px border-base-light"
      data-testid="gridContainer"
    >
      <h2
        class="margin-bottom-0 tablet-lg:font-sans-l desktop-lg:font-sans-xl margin-top-0"
      >
        title
      </h2>
      <div
        class="grid-row grid-gap"
        data-testid="grid"
      >
        <div
          class="tablet:grid-col"
          data-testid="grid"
        >
          <ul
            class="usa-icon-list--size-md grid-row usa-icon-list"
            data-testid="iconList"
          >
            <li
              class="margin-top-4 tablet-lg:grid-col-6 usa-icon-list__item"
              data-testid="iconListItem"
            >
              <div
                class="usa-icon-list__icon"
                data-testid="iconListIcon"
              >
                <svg
                  aria-hidden="true"
                  class="usa-icon usa-icon text-primary-darker"
                  role="img"
                >
                  <use
                    href="/img.jpg#check_circle"
                  />
                </svg>
              </div>
              <div
                class="tablet:padding-right-7 usa-icon-list__content"
                data-testid="iconListContent"
              >
                <h3
                  class="margin-bottom-2 usa-icon-list__title"
                >
                  list.0.title
                </h3>
                <div
                  class="font-sans-md line-height-sans-4 desktop-lg:line-height-sans-6"
                >
                  list.0.content
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    class="padding-top-4 bg-base-lightest"
  >
    <div
      class="grid-container grid-container padding-y-1 tablet:padding-y-3 desktop-lg:padding-y-6 padding-top-0 tablet:padding-top-0 desktop-lg:padding-top-0"
      data-testid="gridContainer"
    >
      <h2
        class="margin-bottom-0 tablet-lg:font-sans-l desktop-lg:font-sans-xl margin-top-0"
      >
        title
      </h2>
      <div
        class="grid-row grid-gap"
        data-testid="grid"
      >
        <div
          class="tablet:grid-col"
          data-testid="grid"
        >
          <ul
            class="usa-icon-list--size-md grid-row usa-icon-list"
            data-testid="iconList"
          >
            <li
              class="margin-top-4 tablet-lg:grid-col-6 usa-icon-list__item"
              data-testid="iconListItem"
            >
              <div
                class="usa-icon-list__icon"
                data-testid="iconListIcon"
              >
                <svg
                  aria-hidden="true"
                  class="usa-icon text-base"
                  role="img"
                >
                  <use
                    href="/img.jpg#check_circle_outline"
                  />
                </svg>
              </div>
              <div
                class="tablet:padding-right-7 usa-icon-list__content"
                data-testid="iconListContent"
              >
                <h3
                  class="margin-bottom-2 usa-icon-list__title"
                >
                  list.0.title
                </h3>
                <div
                  class="font-sans-md line-height-sans-4 desktop-lg:line-height-sans-6"
                >
                  list.0.content
                </div>
              </div>
            </li>
          </ul>
          <h3>
            <a
              class="usa-link--external"
              href="https://github.com/HHS/simpler-grants-gov/issues?q=is%3Aissue%20type%3ADeliverable%20"
              target="_blank"
            >
              link
            </a>
          </h3>
        </div>
      </div>
    </div>
  </div>
  <div
    class="grid-container padding-y-1 tablet:padding-y-3 desktop-lg:padding-y-6  "
    data-testid="gridContainer"
  >
    <div
      class="grid-row grid-gap"
      data-testid="grid"
    >
      <div
        class="grid-row grid-gap-lg"
        data-testid="grid"
      >
        <div
          class="tablet-lg:grid-col-6"
          data-testid="grid"
        >
          <h3
            class="tablet-lg:font-sans-lg tablet-lg:margin-bottom-05"
          >
            involved.title_1
          </h3>
          <p
            class="font-sans-md line-height-sans-4 desktop-lg:line-height-sans-6"
          >
            involved.paragraph_1
          </p>
        </div>
        <div
          class="tablet-lg:grid-col-6"
          data-testid="grid"
        >
          <h3
            class="tablet-lg:font-sans-lg tablet-lg:margin-bottom-05"
          >
            involved.title_2
          </h3>
          <p
            class="font-sans-md line-height-sans-4 desktop-lg:line-height-sans-6"
          >
            involved.paragraph_2
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
